stub-pthreads/stub-pthreads-emulated.c

```diff
@@ -0,0 +1,40 @@
+// This file is linked into wasi-emulated-pthread
```

Comment:
For other stubs that are put in separate archives the respective header file puts all definitions behind a `#define` to prevent accidental usage. For example `_WASI_EMULATED_MMAN` for `sys/mman.h`. All current ones error out with a suggestion to set the right define and add the right staticlib when included, but for pthreads.h this wouldn't really be possible as it contains a mix of fully supported functions and those that merely have a stub. Maybe it would be possible to define the stub functions as macros that expand to `#error` when used?

Reply:
I just tried implementing this, but it's surprisingly tricky to get right. First of all, macros can't expand to other preprocessor directives, so you can't do what you've described. You can expand to `_Static_assert` instead.

For some reason, this ends up making a mess of `expected/*/predefined-macros.txt`, but that might be acceptable.

In any case, I have indeed implemented this suggestion